source/patterns/@aws-solutions-constructs/aws-openapigateway-lambda/README.md

```diff
@@ -0,0 +1,131 @@
+from aws_cdk import (
+    aws_lambda as _lambda,
+    Stack
+)
+from constructs import Construct
+
+ApiGatewayToLambda(self, 'ApiGatewayToLambdaPattern',
+  open_api_spec_input_bucket='s3-bucket-that-holds-openapi-spec-file',
+  open_api_spec_input_key='s3-object-key-of-openapi-spec-file',
+  lambda_function_props=_lambda.FunctionProps(
+      runtime=_lambda.Runtime.PYTHON_3_9,
+      handler='index.handler',
+      code=_lambda.Code.from_asset('lambda')
+  )
+)
+```
+
+Java
+``` java
+import software.constructs.Construct;
+
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.services.lambda.*;
+import software.amazon.awscdk.services.lambda.Runtime;
+import software.amazon.awsconstructs.services.openapigatewaylambda.*;
+
+new ApiGatewayToLambda(this, "ApiGatewayToLambdaPattern", new ApiGatewayToLambdaProps.Builder()
+        .openApiSpecInputBucket("s3-bucket-that-holds-openapi-spec-file")
+        .openApiSpecInputKey("s3-object-key-of-openapi-spec-file")
+        .lambdaFunctionProps(new FunctionProps.Builder()
+          .runtime(Runtime.NODEJS_18_X)
+          .code(Code.fromAsset("lambda"))
+          .handler("index.handler")
+          .build())
+        .build());
+```
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|existingLambdaObj?|[`lambda.Function`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html)|Existing instance of Lambda Function object, providing both this and `lambdaFunctionProps` will cause an error.|
+|lambdaFunctionProps?|[`lambda.FunctionProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.FunctionProps.html)|User provided props to override the default props for the Lambda function.|
+|apiGatewayProps?|[`apigateway.RestApiBaseProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.RestApiBaseProps.html)|Optional user-provided props to override the default props for the API.|
+|openApiSpecInputBucket|`string`|S3 Bucket that holds the input OpenAPI spec file.|
+|openApiSpecInputKey|`string`|S3 Object Key of the input OpenAPI spec file.|
+|openApiSpecOutputBucket?|`string`|Optional S3 Bucket that holds the output (transformed) OpenAPI spec file that will be the input to the SpecRestApi. Defaults to the bucket specified in the `openApiSpecInputBucket` required property.|
+|openApiSpecOutputKey|`string`|Optional S3 object key of the output OpenAPI spec file. Defaults to `openapi.spec`.|
+|openApiSpecUriPlaceholder|`string`|Optional placeholder string that will be overwritten with the actual uri at deploy time. Defaults to `URI_PLACEHOLDER`. For example, if the openapi spec uses the `URI_PLACEHOLDER` string, it will be automatically transformed to: `arn:${partition}:apigateway:${region}:lambda:path/2015-03-31/functions/${lambdaProxyArn}/invocations`.|
```

Comment:
I'm just brainstorming, this may be a terrible idea, but what if we scoped out a namespace for our substitution tokens, then broke them down by type and ordinal:

${constructs.function.0}
${constructs.function.1}  // addresses the idea of accepting an array of lambda functions
${constructs.container.0}
${constructs.stream.0}